vignettes/kaplan_meier.Rmd

```diff
@@ -56,30 +76,29 @@ Censored (0) refers to those subjects who did not experience the event during th
 ### Example
 
 ```{r}
-data <- data.frame(
-  id = 1:5,
-  date_of_death = as.Date(c('2023-05-01', '2023-07-15', NA, '2023-09-10', '2023-06-20')),
-  date_of_relapse = as.Date(c('2023-04-10', NA, '2023-06-01', '2023-08-01', NA)),
-  date_of_last_visit = as.Date(c('2023-04-15', '2023-06-28', '2023-06-10', '2023-08-01', '2023-06-15')),
-  date_first_visit = as.Date(c('2023-01-15', '2023-05-20', '2023-03-01', '2023-07-05', '2023-02-10')),
-  # Event column: 1 for death or relapse, 0 for no event
-  event = c(1, 0, 1, 1, 0)
-)
-
-# Print the data frame
-print(data)
-
-
-ds=data %>%
-  mutate(status= ifelse(event=="death", 1, 0 )) %>%
+# data <- data.frame(
+#   id = 1:5,
+#   date_of_death = as.Date(c('2023-05-01', '2023-07-15', NA, '2023-09-10', '2023-06-20')),
+#   date_of_relapse = as.Date(c('2023-04-10', NA, '2023-06-01', '2023-08-01', NA)),
+#   date_of_last_visit = as.Date(c('2023-04-15', '2023-06-28', '2023-06-10', '2023-08-01', '2023-06-15')),
+#   date_first_visit = as.Date(c('2023-01-15', '2023-05-20', '2023-03-01', '2023-07-05', '2023-02-10')),
+#   # Event column: 1 for death or relapse, 0 for no event
+#   event = c(1, 0, 1, 1, 0)
+# )
+# 
+# # Print the data frame
+# print(data)
+
+data_surv_v2=data_surv %>%
+  mutate(status= ifelse(death==1, 1, 0 )) %>%
   mutate(status_date = case_when(
-    status == 1  ~ pmin(date_of_relapse,date_of_death, na.rm = T), 
+    status == 1  ~ pmin(date_progression, na.rm = T), 
     status == 0  ~ pmax(date_of_last_visit,na.rm = T)
   )) %>% 
   mutate(time_days=status_date-date_first_visit ) %>%  
-  mutate(time_years=time_days/365.25 )  
+  mutate(time_months=time_days/30.5 )  
 
-print(ds)
+head(data_surv_v2)
```

Comment:
`print()` is not necessary, you can just write `ds` there.
using `head()` might remove a bit of clutter but that's totally up to you.